### PR TITLE
Feat/following backend

### PR DESCRIPTION
**Pull request by jeromes123**
**OVERVIEW**
Created the model, controller, and service for relationships. 
Created functions for following and unfollowing users that use softdeletes.

### CODE REVIEW

File: src/app/Services/RelationshipService.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+namespace App\Services;
+
+use App\Models\Relationship;
+class RelationshipService
+{
+    public function follow(int $userToFollowId): void
+    {
+        $existingRelationship = Relationship::where('user_id_follower', auth()->user()->id)
+            ->where('user_id_following', $userToFollowId)
+            ->onlyTrashed()
```
Comment (mikco-hipe) [inline src/app/Services/RelationshipService.php]: I wonder why you're querying with the deleted records?

Comment (jeromes123) [reply src/app/Services/RelationshipService.php]: I modified the code to use status column instead.

**Comment by jeromes123** [comment]
Test for relationship - follow and unfollow

https://github.com/XPERIA679/microblog-ojt1/assets/123230015/d8a17e7e-b94e-44e8-b0c5-86a671277d37

**Review by mikco-hipe — APPROVED** [review]
LGTM 👍

**Pull request merged**